.clang-format

```
@@ -0,0 +1,40 @@
+BasedOnStyle: WebKit
+AccessModifierOffset: -1
+AlignAfterOpenBracket: Align
+AlignConsecutiveAssignments: Consecutive
+AlignConsecutiveDeclarations: Consecutive
+AlignEscapedNewlines: DontAlign
+AlignOperands: AlignAfterOperator
+AlignTrailingComments: true
+AllowShortEnumsOnASingleLine: false
+AllowShortCaseLabelsOnASingleLine: true
+AllowShortIfStatementsOnASingleLine: false
+AllowShortLoopsOnASingleLine: true
+AlwaysBreakTemplateDeclarations: No

```

```suggestion
AlwaysBreakTemplateDeclarations: Yes
```
85 of 112 cases have the line break on master.

---

> 85 of 112 cases have the line break on master.

unfortunatly I don't think the PR goal was to near the master style code at all, even though that would have left less problems to solve.

---

Thats not entirely true, the entire clang file was generated using the current code base as the input for the clang file.
https://github.com/official-stockfish/Stockfish/issues/3608#issuecomment-1717998038

The settings of a few options that are only occuring a few times is not a show stopper and can be easily changed. 
If you create a minimal config file which only indents the code, you will already have the if conditions not aligned anymore.

The main idea from vondele/me was to propose the new clang-format and see what the opinion's are. 
Of course there are always a few things which person a doesnt like but person b, but in a lot of these cases we just have to bite the bullet between one of those. 
While certain formatting options are definitely not in the codebase like the alignment, I think they visually improve the code readability. Though I havent gone through each of them individually. 
This clang-format file was simply trying to be an improvement over the one provided in the linked issue, which vondele found not so good. 

In general I think the formatted code is looking pretty readable (apart from maybe a few weird array formats or conditionals) and adding a formatter, always eases the DX and helps new users.
Obviously there will be major formatting changes but there's no way around that.

State of the PR:
I think we should try to keep some of array initializers and conditional alignments (help appreciated) and get @snicolet on board.